crates/sui-cluster-test/src/lib.rs

```diff
@@ -83,20 +83,20 @@ impl TestContext {
 
     pub async fn setup(options: ClusterTestOpt) -> Result<Self, anyhow::Error> {
         let cluster = ClusterFactory::start(&options).await?;
-        let faucet_url = cluster.faucet_url().map(String::from);
         let wallet_client = WalletClient::new_from_cluster(&cluster).await;
+        let faucet = FaucetClientFactory::new_from_cluster(&cluster).await;
         Ok(Self {
             cluster,
             client: wallet_client,
-            faucet: FaucetClientFactory::create(&options, faucet_url),
+            faucet,
         })
     }
 
     // TODO: figure out a more efficient way to test a local cluster
     // A potential way to do this is to subscribe to txns from fullnode
     // when the feature is ready
     pub async fn let_fullnode_sync(&self) {
-        let duration = Duration::from_secs(10);
+        let duration = Duration::from_secs(5);
```

Comment:
Is there value in making the sleep duration a function arg user can pass in?

Reply:
probably not at the moment. We are making it very generous at 5 seconds at the moment for simplicity. But this is fairly easy to change if we see need